File: API-Server/Happy Habits App/Forms/SleepStatistics.cs
using System.Text.Json.Serialization;

namespace Happy_Habits_App.Forms
{
    public class SleepStatistics(List<float> sleepDurations, int dailyAverageHours, int dailyAverageMinutes, int differenceInHours, int differenceInMinutes, string mostFrequentQuality)
    {
        [JsonPropertyName("sleepDurations")]
        public List<float> SleepDurations { get; set; } = sleepDurations;

        [JsonPropertyName("dailyAverageHours")]
        public int DailyAverageHours { get; set; } = dailyAverageHours;
        [JsonPropertyName("dailyAverageMinutes")]
        public int DailyAverageMinutes { get; set; } = dailyAverageMinutes;
        public int DifferenceInHours { get; set; } = differenceInHours;
        [JsonPropertyName("differenceInMinutes")]
        public int DifferenceInMinutes { get; set; } = differenceInMinutes;
        [JsonPropertyName("mostFrequentQuality")]
        public string MostFrequentQuality { get; set; } = mostFrequentQuality;
    }
}

File: Mobile-App/app/src/main/java/com/example/happyhabits/feature_application/feauture_sleep/data/model/SleepStatistics.kt
package com.example.happyhabits.feature_application.feauture_sleep.data.model

data class SleepStatistics (
    val sleepDurations : List<Float>,
    val dailyAverageHours : Float,
    val dailyAverageMinutes : Float,
    val differenceInHours : Float,
    val differenceInMinutes : Float,
    val mostFrequentQuality : String
)

